### ISSUE TITLE
Image name and version should be available through Rancher Metadata Service

### ISSUE THREAD

**Issue by greatvovan** [rancher/rancher#11799]
[Rancher Metadata Service](http://rancher.com/docs/rancher/v1.6/en/rancher-services/metadata-service/) provides a convenient way to explore the service itself as well as other services and stacks.

However the service lacks the possibility (API) to get Docker image name and version (URL) for a service or container. This information can only be found in GUI.

The reason why this might be an extremely useful feature, is the fact that another service's API might change from one version to another, and being aware of version might help client service a lot. Knowing own image version is also a useful feaure for logging purposes.

To sum up, we need to augment Rancher Metadata Service API with folowing:

    http://rancher-metadata/latest/service/image
    http://rancher-metadata/latest/container/image
    http://rancher-metadata/latest/services/my-service/image
    http://rancher-metadata/latest/services/my-service/containers/0/image

Workarounds are appreciated.

**Comment by vincent99** [comment]
Metadata is mainly for things like service/ip/port discovery and leader election.  It is not the API, and it is a long slippery slope down adding every field someone might possibly want from the resource into metadata.

If you want info from the API you can talk to the API:  http://rancher.com/docs/rancher/v1.6/en/rancher-services/service-accounts/

**Comment by greatvovan** [comment]
For the purpose I wrote about, Rancher API is also insufficient.
The API does not have documented methods to get service's image URL (I hope to be wrong but I did not find anything). The only methods that appear to return image info are ones like:

    http://rancher/v2-beta/projects/1a5/services?limit=-1&sort=name

that seem to be a part of Rancher's web UI.
But they return HUGE and scary JSON with all services in infrastructure. One should iterate over all of them to match known service/stack names. Compare this to:

    http://rancher-metadata/latest/self/service/image

**Comment by digitalbanana** [comment]
I would also suggest that image name and tag would be extremely useful for our DevOps platform. We are working on integrating Rancher into our DevOpos stack such that we can present a dashboard of exactly what is deployed where. The point about slippery slopes is a good one but I would think the the image, as the basic unit of deployment should be available via the metadata service

**Comment by loganhz** [comment]
Thanks for your report!

With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.

If you think we should keep this issue open, please let me know.
